### PR TITLE
[client/catapult] fix: upgrade to openssl 3.1.0 

### PR DESCRIPTION
**Pull request by Wayonb**
## What's the issue?
catapult client is using an older version of openssl

## How have you changed the behavior?
upgrade openssl to version v3.1.0

## How was this change tested?
ran the catapult tests on my dev box.

### CODE REVIEW

File: client/catapult/src/catapult/crypto/OpensslContexts.cpp [hunk 1]
```diff
@@ -37,15 +37,11 @@ namespace catapult { namespace crypto {
 	// region OpensslDigestContext
 
 	OpensslDigestContext::OpensslDigestContext() {
-		std::memset(&m_buffer, 0, CountOf(m_buffer));
-	}
-
-	OpensslDigestContext::~OpensslDigestContext() {
-		reset();
+		m_buffer = UniquePtr(EVP_MD_CTX_new(), [](context_type* p) { EVP_MD_CTX_free(p); });
```
Comment (Jaguar0625) [inline client/catapult/src/catapult/crypto/OpensslContexts.cpp:40]: this is doing an allocation - that defeats the purpose of these contexts to be stack allocated 😅

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]

File: client/catapult/src/catapult/crypto/OpensslContexts.cpp (hunk 1 above)
Comment (Wayonb) [reply client/catapult/src/catapult/crypto/OpensslContexts.cpp:40]: haha, I was wondering this.  I guess you want stack allocation for performance?

Comment (Jaguar0625) [reply client/catapult/src/catapult/crypto/OpensslContexts.cpp:40]: yep, would prefer to keep that, if possible

**Review by Jaguar0625 — APPROVED** [review]
👍

**Pull request merged**